src/dynamics/ads_player/ads_player.js

```diff
@@ -386,11 +386,28 @@ Scoped.define("module:Ads.Dynamics.Player", [
                     }
                     return true;
                 },
+                getMediaUrl: function(adObj) {
+                    let adMediaUrl = adObj?.data?.mediaUrl;
+                    if (adMediaUrl) {
+                        return adMediaUrl;
+                    }
+                    // backup plan if mediaurl is not in data object
+                    const mediaFiles = adObj?.data?.traffickingParameters && JSON.parse(adObj.data.traffickingParameters)?.mediaFiles;
```

Comment:
hey @EdmundsBankai, all looks great, just I think here should be `||` not `&&` (if first one is main source) as on this way it will return undefined if second condition will be `undefined` (I think it's well works on React JSX part). Also for `JSON.parse` always will be cover it with some try/catch as wrong argument passed to JSON.parser will throw an error.

Reply:
added try catch and wrapped the block in

```
if (adObj?.data?.traffickingParameters) {
   ... JSON parsing logic
}
```

instead